### ISSUE TITLE
Identifier Validation Challenge- SimpleHTTPS- should verify that Index is NOT allowed in .well-known/ path

### ISSUE THREAD

**Issue by libove** [letsencrypt/acme-spec#14]
To avoid aggressive timing attacks, the Server should validate that Indexing (Directory Listing) is NOT allowed in the .well-known/ path under which it has been requested by the Client to find the Client's proof of SimpleHTTPS Identifier Validation (a file containing the nonce sent by the Server). 

I theorize that an aggressive attacker could repeatedly attempt to index the Client website, and if the Client website permits directory browsing, then the attacker could find the nonce in the .well-known/ subpath.

In theory, a nonce is usable only once, and the real Client will immediately use this nonce in its response to the Server to prove the Client's claim, making a timing attack unlikely.

If the ACME protocol design or a Server's implementation would permit the attacker-Client, now having the Server's nonce, to authenticate as the Client, this might be a real threat.

In any case, it's a useful extra security step to help the Client administrator avoid common configuration errors such as allowing directory browsing in places where it should not be allowed, which often result in security breaches. (Although this could be preferred to be kept out of the ACME protocol design, as being out of scope).

If I've missed something totally ignorant and my concern or suggestion here are completely useless, I'll take my lumps, and I apologise in advance!

**Comment by bifurcation** [comment]
This is addressed by the general MitM resistance measures in #156
